pipe_vms_research/utils/bqtools.py

```diff
@@ -0,0 +1,214 @@
+            sys.exit(1)
+        except AlreadyExistErr as err:
+            logger.warn(f'Already exists table: {err}.')
+        except Exception as err:
+            logger.error(f'Unrecongnized error: {err}.')
+            sys.exit(1)
+
+    def update_table(self, destination_table, description, schema):
+        """Updates the schema of an existent table.
+
+        :param destination_table: dataset.table of BQ.
+        :type destination_table: str.
+        :param description: the main description of the table.
+        :type description: str.
+        :param schema: the schema of the table.
+        :type schema: list[bigquery.SchemaField].
+        """
+        destination_table_ds, destination_table_tb = destination_table.split('.')
+        destination_dataset_ref = bigquery.DatasetReference(self.bq_client.project, destination_table_ds)
+        destination_table_ref = destination_dataset_ref.table(destination_table_tb)
+        try:
+            table = self.bq_client.get_table(destination_table_ref)
+            table.schema = schema
+            table.description = description
+            result = self.bq_client.update_table(table, ["description","schema"])
+            logger.info(f'Update table schema from table {destination_table_ds}.{destination_table_tb}. Result: {result}')
+        except BadRequest as err:
+            logger.error(f'update_table - Bad request received {err}.')
+            sys.exit(1)
+        except Exception as err:
+            logger.error(f'update_table - Unrecongnized error: {err}.')
+            sys.exit(1)
+
+    def update_table_descr(self, destination_table, description):
+        """Updates the schema of an existent table.
+
+        :param destination_table: dataset.table of BQ.
+        :type destination_table: str.
+        :param description: the main description of the table.
+        :type description: str.
+        """
+        destination_table_ds, destination_table_tb = destination_table.split('.')
+        destination_dataset_ref = bigquery.DatasetReference(self.bq_client.project, destination_table_ds)
+        destination_table_ref = destination_dataset_ref.table(destination_table_tb)
+        try:
+            table = self.bq_client.get_table(destination_table_ref)
+            table.description = description
+            result = self.bq_client.update_table(table, ["description"])
+            logger.info(f'Update table description from table {destination_table_ds}.{destination_table_tb}. Result: {result}')
+        except BadRequest as err:
+            logger.error(f'update_table_descr - Bad request received {err}.')
+            sys.exit(1)
+        except Exception as err:
+            logger.error(f'update_table_descr - Unrecongnized error: {err}.')
+            sys.exit(1)
+
+    def run_estimation_query(self, query, destination, labels, is_partitioned:bool=True):
+        return self.run_query(query, destination, labels, True, is_partitioned)
```

Comment:
As I said in the other comment, I would remove this feature and all it's code altogether because it's so unreliable on monthly partitioned tables. But even if we don't and we want to keep this, having the `run_query` method implement this via a flag is not a good idea. I know that internally it's the same BQ API call, but the semantics of the method are very different, and I don't like a single flag changing what something's doing so much (and it also causes your method to have a bunch of `if flag:` everywhere).